pkg/slayers/path/scion/raw_spec.gobra

```diff
@@ -659,6 +699,187 @@ func (s *Raw) LastHopLemma(ubuf []byte) {
 	numINF := NumInfoFields(seg1Len, seg2Len, seg3Len)
 	offset := HopFieldOffset(numINF, 0, MetaLen)
 	pkt := reveal s.absPkt(ubuf)
-	assert pkt.CurrSeg == reveal CurrSeg(ubuf, offset+prevSegLen, currINFIdx, currHFIdx-prevSegLen, segLen, MetaLen)
+	assert pkt.CurrSeg == reveal CurrSeg(ubuf, offset + path.HopLen * prevSegLen, currINFIdx, currHFIdx-prevSegLen, segLen, MetaLen)
 	assert len(pkt.CurrSeg.Future) == 1
+}
+
+ghost
+preserves acc(s.Mem(ubuf), R55)
+preserves s.GetIsXoverSpec(ubuf)
+preserves acc(sl.AbsSlice_Bytes(ubuf, 0, len(ubuf)), R56)
+preserves validPktMetaHdr(ubuf)
+preserves s.EqAbsHeader(ubuf)
+ensures  s.absPkt(ubuf).LeftSeg != none[io.IO_seg2]
+ensures len(s.absPkt(ubuf).CurrSeg.Future) == 1
+decreases
+func (s *Raw) XoverLemma(ubuf []byte) {
+	reveal validPktMetaHdr(ubuf)
+	metaHdr := RawBytesToMetaHdr(ubuf)
+	currINFIdx := int(metaHdr.CurrINF)
+	currHFIdx := int(metaHdr.CurrHF)
+	seg1Len := int(metaHdr.SegLen[0])
+	seg2Len := int(metaHdr.SegLen[1])
+	seg3Len := int(metaHdr.SegLen[2])
+	segLen := LengthOfCurrSeg(currHFIdx, seg1Len, seg2Len, seg3Len)
+	prevSegLen := LengthOfPrevSeg(currHFIdx, seg1Len, seg2Len, seg3Len)
+	numINF := NumInfoFields(seg1Len, seg2Len, seg3Len)
+	offset := HopFieldOffset(numINF, 0, MetaLen)
+	pkt := reveal s.absPkt(ubuf)
+	assert pkt.CurrSeg == reveal CurrSeg(ubuf, offset + path.HopLen * prevSegLen, currINFIdx, currHFIdx-prevSegLen, segLen, MetaLen)
+	assert pkt.LeftSeg == reveal LeftSeg(ubuf, currINFIdx + 1, seg1Len, seg2Len , seg3Len, MetaLen)
+	assert len(pkt.CurrSeg.Future) == 1
+	assert pkt.LeftSeg != none[io.IO_seg2]
+}
+
+ghost
+requires  path.InfoFieldOffset(currINFIdx, 0) + path.InfoLen <= offset
+requires  0 < segLen
+requires  offset + path.HopLen * segLen <= len(raw)
+requires  0 <= currHFIdx && currHFIdx < segLen
+requires  0 <= currINFIdx && currINFIdx < 3
+preserves acc(sl.AbsSlice_Bytes(raw, 0, len(raw)), R56)
+ensures   len(CurrSeg(raw, offset, currINFIdx, currHFIdx, segLen, 0).Future) > 0
+decreases
+func CurrSegFutureLenLemma(raw []byte, offset int, currINFIdx int, currHFIdx int, segLen int) {
+	reveal CurrSeg(raw, offset, currINFIdx, currHFIdx, segLen, 0)
+}
+
+ghost
+requires  0 < seg1Len
+requires  0 <= seg2Len
+requires  0 <= seg3Len
+requires  pktLen(seg1Len, seg2Len, seg3Len, 0) <= len(raw)
+requires  0 <= currINFIdx && currINFIdx < 2
+requires  0 <= currINFIdx ==> 0 < seg2Len
+requires  1 <= currINFIdx ==> 0 < seg3Len
```

Comment:
This can be simplified: because `0 <= currINFIdx && currINFIdx < 2`, then the LHS of `0 <= currINFIdx ==> 0 < seg2Len` is trivially true, so we can simplify it to `0 < seg2Len`.

Also, `currINFIdx` would be better as `currInfIdx`

Reply:
I changed `currINFIdx` to `currInfIdx` throughout the entire codebase for consistency.